src/test/java/com/google/api/generator/gapic/protoparser/RoutingRuleParserTest.java

```diff
@@ -0,0 +1,109 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.api.generator.gapic.protoparser;
+
+import static com.google.common.truth.Truth.assertThat;
+import static org.junit.Assert.assertThrows;
+
+import com.google.api.generator.gapic.model.Message;
+import com.google.api.generator.gapic.model.RoutingHeaders;
+import com.google.api.generator.gapic.model.RoutingHeaders.RoutingHeader;
+import com.google.explicit.dynamic.routing.header.RoutingRuleParserTestingOuterClass;
+import com.google.protobuf.Descriptors.FileDescriptor;
+import com.google.protobuf.Descriptors.MethodDescriptor;
+import com.google.protobuf.Descriptors.ServiceDescriptor;
+import java.util.Map;
+import org.junit.Test;
+
+public class RoutingRuleParserTest {
+
+  private static final FileDescriptor TESTING_FILE_DESCRIPTOR =
+      RoutingRuleParserTestingOuterClass.getDescriptor();
+  private static final Map<String, Message> MESSAGES =
```

Comment:
`Map` is mutable.

Reply:
Refactored `MESSAGES` not to be a static final field.